cli/azd/pkg/project/service_target_aks.go

```diff
@@ -225,6 +264,194 @@ func (t *aksTarget) Deploy(
 		})
 }
 
+// deployManifests deploys raw or templated yaml manifests to the k8s cluster
+func (t *aksTarget) deployManifests(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	task *async.TaskContextWithProgress[*ServiceDeployResult, ServiceProgress],
+) (*kubectl.Deployment, error) {
+	deploymentPath := serviceConfig.K8s.DeploymentPath
+	if deploymentPath == "" {
+		deploymentPath = defaultDeploymentPath
+	}
+
+	// Manifests are optional so we will continue if the directory does not exist
+	if _, err := os.Stat(deploymentPath); os.IsNotExist(err) {
+		return nil, err
+	}
+
+	task.SetProgress(NewServiceProgress("Applying k8s manifests"))
+	err := t.kubectl.Apply(
+		ctx,
+		filepath.Join(serviceConfig.RelativePath, deploymentPath),
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed applying kube manifests: %w", err)
+	}
+
+	deploymentName := serviceConfig.K8s.Deployment.Name
+	if deploymentName == "" {
+		deploymentName = serviceConfig.Name
+	}
+
+	// It is not a requirement for a AZD deploy to contain a deployment object
+	// If we don't find any deployment within the namespace we will continue
+	task.SetProgress(NewServiceProgress("Verifying deployment"))
+	deployment, err := t.waitForDeployment(ctx, deploymentName)
+	if err != nil && !errors.Is(err, kubectl.ErrResourceNotFound) {
+		return nil, err
+	}
+
+	return deployment, nil
+}
+
+// deployKustomize deploys kustomize manifests to the k8s cluster
+func (t *aksTarget) deployKustomize(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	task *async.TaskContextWithProgress[*ServiceDeployResult, ServiceProgress],
+) (bool, error) {
+	if serviceConfig.K8s.Kustomize == nil {
+		return false, nil
+	}
+
+	if !t.featureManager.IsEnabled(featureKustomize) {
+		return false, fmt.Errorf(
+			"Kustomize support is not enabled. Run '%s' to enable it.",
+			alpha.GetEnableCommand(featureKustomize),
+		)
+	}
```

Comment:
Should we just print a warning instead of returning an error?
We could say that `kustomize` configuration detected but the feature is off so it is ignored.  

That would allow folks to turn the feature off and keep running the project w/o getting an error.

Same for Helm

Reply:
This is the existing experience precedence we have already set for alpha features in `azd`.  If the feature is disabled, we throw an error and tell them how to enable it and force them to run again.

If we want to change this precedence I'm happy to revisit all the alpha features in a separate PR.